lib/actions/api.js

```diff
@@ -321,14 +305,49 @@ export function getRoutingParams (query, config, ignoreRealtimeUpdates) {
       delete params.endTimeValid
     }
   }
+}
 
-  // TODO: check that valid from/to locations are provided
+function paramIsDefined (query, key) {
+  return key in query && query[key] !== '' && query[key] !== null
+}
 
-  // hack to add walking to driving/TNC trips
-  if (hasCar(params.mode)) {
-    params.mode += ',WALK'
-  }
+/**
+ * A given parameter is included in the request if all of the following:
```

Comment:
```suggestion
 * A given parameter is included in the request if all of the following conditions are met:
```

Reply:
Evan pointed out that getRoutingParams is already exported from otp-ui, so I'm removing this code in favor of that version of the method.